academy/0.0-B9lab-Blockchains/4_consensus.md

```diff
@@ -0,0 +1,112 @@
+
+Many consider DPoS algorithms superior to PoW and PoS because of their fast block creation, a high degree of security, energy efficiency, level of integrity, and democratic structure. However, DPoS systems are less decentralized than PoW and PoS systems because they have fixed validator sets and higher barriers to entry.
+
+## Practical Byzantine Fault Tolerance (PBFT)
+
+PBFT was published in 1999 and arose from academia. It is a three-phase protocol, in which the client sends a request to a so-called primary. 
+
+In the first phase, the primary broadcasts the request with a sequence number to the replicas. Then the replicas agree on the sequence number and create a message. 
+
+When a threshold of agreement is reached by the replicas, which is akin to sufficient co-signers, the message is verified. The replicas agree on the overall order of transactions within a view. The broader network is informed about transaction blocks when they are finalized. That is to say, signed by sufficient replicas to be sure that the block is finalized. 
+
+This of course is a very simplified presentation.
+
+<HighlightBox type="reading">
+
+    You can access the paper by Miguel Castro and Barabara Liskov [here](http://pmg.csail.mit.edu/papers/osdi99.pdf). There is also an intuitive [presentation](http://www.comp.nus.edu.sg/~rahul/allfiles/cs6234-16-pbft.pdf).
+
+</HighlightBox>
+
+## Tendermint ##
+
+While most consensus implementations are tightly coupled with a particular blockchain project, Tendermint is a toolkit that focuses on simplifying the process of creating blockchains with different properties and it includes Tendermint consensus. As the reader might guess, Cosmos relies on Tendermint consensus which offers the most mature Byzantine Fault Tolerance, transaction finality as well as a great deal of flexibility - flexibility that is passed through to the creators of custom blockchains built with the Cosmos SDK. 
+
+[Tendermint](./1-what-is-cosmos/blockchain-and-cosmos.md) will be explored in detail later on. 
+
+## Many others ##
+
+Many teams have experimented with distributed consensus using many algorithms deployed to various chains. In the context of blockchain technology, "consensus" refers to a method of ordering transactions. The many algorithms that exist can be compared in terms of the distribution of authority, expected performance and transaction finality. Transaction finality can be categorized as "probablistic" and "deterministic", i.e. is it exponentially more improbable that a seen transaction will be reversed, or does the protocol prevent alterations to history in a definite way. 
+
+The above are the most popular by usage as well as the most important to keep in mind when diving into consensus in Cosmos. 
+
+<HighlightBox type="reading">
```

Comment:
I tailored the reading list and deleted all readings that seem to not be completely helpful for those diving into Cosmos and added some readings that would fit in my opinion. Please recheck.

Reply:
fantastic. thank you.